driver-sync/src/main/com/mongodb/client/internal/ChangeStreamIterableImpl.java

```diff
@@ -114,11 +118,19 @@
     @Override
     public <TDocument> MongoIterable<TDocument> withDocumentClass(final Class<TDocument> clazz) {
         return new MongoIterableImpl<TDocument>(getClientSession(), getExecutor(), getReadConcern(), getReadPreference(), getRetryReads()) {
-            private ReadOperation<BatchCursor<TDocument>> operation = createChangeStreamOperation(codecRegistry.get(clazz));
+            @Override
+            public MongoCursor<TDocument> iterator() {
+                return cursor();
+            }
+
+            @Override
+            public MongoChangeStreamCursor<TDocument> cursor() {
+                return new MongoChangeStreamCursorImpl<TDocument>(execute(), codecRegistry.get(clazz), initialResumeToken());
+            }
 
             @Override
             public ReadOperation<BatchCursor<TDocument>> asReadOperation() {
-                return operation;
+                throw new UnsupportedOperationException();
```

Comment:
Can we avoid throwing this exception here and in similar places? (It's difficult to see it from github diff)

Reply:
Can we just confirm that you can still call `map` on the resulting iterable?

Reply:
I have written a couple of tests that call `map` on the iterable, one with `withDocumentClass()` specified and one without, and the tests succeed.

Reply:
Thanks @jstewart-mongo